replit.nix

```diff
@@ -1,7 +0,0 @@
-{ pkgs }: {
-  deps = [
-    pkgs.nodejs_20
-    pkgs.nodePackages.typescript-language-server
-    pkgs.nodePackages.pnpm
```

Comment:
when CodeSandbox first starts, it uses the (old) pnpm v8 version, which ignores the lockfile:

![Screenshot 2024-05-06 at 16 17 18](https://github.com/upleveled/express-guest-list-api-memory-data-store/assets/1935696/e350f365-b16c-45ab-a692-ab78cd373b96)

is there an easy way to get CodeSandbox to use pnpm v9? eg. with `"engines"` in `package.json`?

Reply:
Added `"pnpm": ">=9.0.0"` but CodeSandbox failed with this error message

```
 ERR_PNPM_UNSUPPORTED_ENGINE  Unsupported environment (bad pnpm and/or Node.js version)

Your pnpm version is incompatible with "/workspace".

Expected version: >=9.0.0
Got: 8.15.6

This is happening because the package's manifest has an engines.pnpm field specified.
To fix this issue, install the required pnpm version globally.

To install the latest version of pnpm, run "pnpm i -g pnpm".

To check your pnpm version, run "pnpm -v".
``` 

<img width="596" alt="Screenshot 2024-05-07 at 15 13 43" src="https://github.com/upleveled/express-guest-list-api-memory-data-store/assets/80746311/6bdb4af7-3d13-48ca-9497-3a6efee8cf09">

Reply:
ok, hm... what about a `.codesandbox/Dockerfile` file like Ives mentioned here:

https://github.com/codesandbox/codesandbox-client/discussions/8111#discussioncomment-7267935

eg. cherry-picking some simple things from [my comment](https://github.com/pnpm/pnpm/issues/784#issuecomment-1518582235), maybe only 2 lines are needed:

```dockerfile
FROM node:20-alpine
RUN corepack enable
```

Reply:
Created a `.codesandbox/Dockerfile`, and it seems to work. I can't see the Dockerfile build, but when forking the repo on CodeSandbox, I can see that the Devbox is using the latest pnpm version. Will test again in my [forked repo](https://github.com/ProchaLu/express-guest-list-api-memory-data-store)
```
➜  /workspace git:(main) pnpm -v
9.1.0
```

Reply:
> I can see that the Devbox is using the latest pnpm version

be careful here - the goal is not to run "the latest version". it's to run the version in the `"engines"` in `package.json` (which is still not specified in this PR)

Reply:
Changed to use `yq` to get the version number from `engines` in `package.json` 
```
RUN corepack enable && corepack prepare pnpm@$(yq e '.engines.pnpm' package.json)
```

Commit [2c880f9](https://github.com/upleveled/express-guest-list-api-memory-data-store/pull/294/commits/2c880f98a976b488f4a7c5319c0a07faf500607e)

Reply:
- that commit also added the `CI` environment variable - I was suggesting `CI` as an alternative to `corepack prepare ...` commands - I don't think you need both
- that commit doesn't have `--activate`, does Corepack need that?

Reply:
- `corepack prepare pnpm@latest`, is the command that downloads and prepares the latest version of `pnpm`, and it doesn't set it as the default version. If there is a pre-existing active `pnpm` version, that one remains in use.

- `corepack prepare pnpm@latest --activate`: This command not only prepares the latest version of `pnpm` but also sets it as the default for `pnpm` commands. So `--activate` makes sure all `pnpm` commands use the version we added. 

Currently, we create our own Docker Image on CodeSandbox, and the `pnpm` version we are adding is the default version. Students will not run additional `pnpm` commands immediately, but to avoid future errors, I would use the `--activate` option. 

Commit [c32a304](https://github.com/upleveled/express-guest-list-api-memory-data-store/pull/294/commits/c32a304b0e15ad240375c8771374fbabb1d855e6)